vault/resource_generic_secret.go

```diff
@@ -186,11 +213,35 @@ func genericSecretResourceRead(d *schema.ResourceData, meta interface{}) error {
 			return fmt.Errorf("error marshaling JSON for %q: %s", path, err)
 		}
 
+		d.Set("read_data_json", string(jsonData))
+
+		var relevantData map[string]interface{}
+		if ignore_absent_fields {
+			var suppliedData map[string]interface{}
+			err = json.Unmarshal([]byte(d.Get("data_json").(string)), &suppliedData)
+			if err != nil {
+				return fmt.Errorf("data_json %#v syntax error: %s", d.Get("data_json"), err)
+			}
+			relevantData = suppliedData
```

Comment:
Something about this code doesn't seem right to me. I think it might be that this line isn't needed? I know this PR hasn't reached test-writing stage yet - it would be good to get some coverage on this if we decide to keep it.

Reply:
Yes, because sometimes fields that are written are not returned by the read operation. Initializing relevantData with suppliedData ensures that those fields' absence from the read operation will not cause terraform to consider the resource to be out of date. In my example, the `password` field is an example of this. If you first write to `auth/userpass/users/u1` with the `password` field and the read back `auth/userpass/users/u1`, you get back some fields you didn't write and you don't get back password. This causes password to be persisted in the state, but any user who is using the vault terraform provider must always be mindful of what's being written to state. The use case here would be to use a dummy password that is changed out of band. Let me know if this is clear or if I should clarify further.